### DISCUSSION TITLE
I want to use features like X-Real-IP

### DISCUSSION THREAD

**Discussion by uu5007mp** [tobychui/zoraxy#224] [Q&A]
Can I use the X-Real-IP feature with HTTP proxy? If so, how can I set it up?
(I don't speak English, so I use a translator)

**Comment by tobychui** [comment]
@uu5007mp 

The X-Real-IP headers are automatically injected and sent to origin server (upstreams) with the proxy request. You can overwrite or remove the default value by using the "Custom Header" feature when editing a HTTP proxy rule.

X-Real-IPヘッダーは自動的に注入され、プロキシリクエストとともにオリジンサーバー（アップストリーム）に送られる。HTTPプロキシルールの編集時に "Custom Header "機能を使用することで、デフォルト値を上書きまたは削除することができます。

![圖片](https://github.com/tobychui/zoraxy/assets/24617523/3923b2e3-e1d8-484c-8b0d-b2d24652bcc4)

![圖片](https://github.com/tobychui/zoraxy/assets/24617523/4aec2a54-4c40-4c1d-8c18-8ab54e071a63)

**Reply by uu5007mp** [reply]
So does that mean I just need to add a header? Or is X-Real-IP already on from the start?
Also, thank you for sending it in Japanese!

**Reply by tobychui** [reply]
@uu5007mp You are correct, X-Real-IP is on from the start and unless you want to disable it or overwrite it with something else. For example, some specific use-cases will require `X-Real-IP: localhost`.

そうです、X-Real-IPは最初からオンになっています。例えば、ある特定の使用例では、`X-Real-IP: localhost`が必要になります。

**Reply by uu5007mp** [reply]
![image](https://github.com/tobychui/zoraxy/assets/86598757/87151041-3579-419b-a0ad-615742155cf1)
だめです...私はDiscordBOTでIPアドレス式認証を作っているのですが内部IPアドレスになります....

It doesn't work... I'm creating IP address authentication with DiscordBOT, but it's an internal IP address...

**Reply by tobychui** [reply]
Is that `192.168.4.3` the IP address of your Zoraxy server? Here is a script for you to help with debugging. You can put it inside a php server like Apache and see if the headers are correctly passed through to your bot. 

`192.168.4.3`はZoraxyサーバーのIPアドレスですか？デバッグに役立つスクリプトがあります。このスクリプトをApacheのようなphpサーバーに設置し、ヘッダーが正しくボットに送信されているかどうかを確認することができます。
[debug.zip](https://github.com/user-attachments/files/16024794/debug.zip)

One important notes is that if your original request already contains the `X-Real-IP` header, Zoraxy will not rewrite it for you. 
重要な注意点として、元のリクエストがすでに `X-Real-IP` ヘッダーを含んでいる場合、Zoraxyはそれを書き換えないということです。

**Reply by uu5007mp** [reply]
Sorry. It's been a long time since I last did this. Below is the image I tried.
すみません。いろいろあってものすごく久しぶりになりました。下がやってみた画像です。
![image](https://github.com/user-attachments/assets/714adb3b-ab50-4611-921a-6a5b5b95aadf)

**Reply by uu5007mp** [reply]
ちなみにBOTのほうはだめそうです
By the way, it seems that BOT is not allowed.
![image](https://github.com/user-attachments/assets/c1aa38b4-b0be-4e55-a477-01655d497376)

**Reply by tobychui** [reply]
@uu5007mp 

It seems you are using Cloudflare and Cloudflare will block bot access by default
Cloudflareを使用しているようですが、Cloudflareはデフォルトでボットからのアクセスをブロックします。

![圖片](https://github.com/user-attachments/assets/dbd9697e-63ad-43b8-8aa6-9b45294abf7c)

The X-Real-IP header value also seems correct
X-Real-IPヘッダーの値も正しいようです。

![圖片](https://github.com/user-attachments/assets/0554a07e-1257-40ce-8a26-38fd75396894)

**Reply by uu5007mp** [reply]
No..It's a Discord bot so it will filter the IP addresses that access the website.
いえ..DiscordbotなのでwebサイトにアクセスしたIPを選別するってことです。
If you can, could you please look at the Discordbot code and check it...?

**Reply by tobychui** [reply]
From what I see, Zoraxy is doing exactly what it should do serving your request with correct IP headers. If you still got issues with the Discord bot, please seek help from the Discord community.

Zoraxyはあなたのリクエストに正しいIPヘッダーを提供しています。それでもDiscordボットに問題がある場合は、Discordコミュニティに助けを求めてください。

**Reply by taslabs-net** [reply]
Do you have this setting turned on in your Cloudflare dashboard?


![image](https://github.com/user-attachments/assets/e62ff44c-133c-43a0-be2f-67de6f499999)
